File: 2025-12-02_(AGC_22.1)/football-club/kaan.cpp
#include <iostream>
#include <vector>

template <typename Comparator>
const int inversions(const std::vector<int>& nums, Comparator cmp) {
    int inv{};
    
    const size_t n{ nums.size() };
    
    for (size_t i = 0; i < n; ++i) {
        for (size_t j = i+1; j < n; ++j) {
            if (cmp(nums[i], nums[j])) {
                ++inv;
            }
        }
    }
    return inv;
}

const int solve(const std::vector<int>& nums) {
    const size_t n{ nums.size() };
    
    const int inv_asc{ inversions(nums, std::greater<>()) };
    const int inv_desc{ inversions(nums, std::less<>()) };
    
    return std::min(inv_asc, inv_desc);
}

int main() {
    size_t n{};
    std::cin >> n;
    
    std::vector<int> nums(n);
    for (size_t i = 0; i < n; ++i) {
        std::cin >> nums[i];
    }
    std::cout << solve(nums) << '\n';
    
	return 0;
}

File: 2025-12-02_(AGC_22.1)/football-club/kaan.py
import operator


n = int(input())
nums = list(map(int, input().split()))

def inversions(lst, op: operator) -> int:
    inv = 0
    n = len(lst)
    for i in range(n):
        for j in range(i+1, n):
            if op(lst[i], lst[j]):
                inv += 1
    return inv
    
def solve(lst) -> int:
    inv_asc = inversions(lst, operator.gt)
    inv_desc = inversions(lst, operator.lt)
    
    return min(inv_asc, inv_desc)
    
print(solve(nums))
    